CMakeLists.txt

```diff
@@ -1,6 +1,6 @@
 cmake_minimum_required(VERSION 3.12)
 project(pyg)
-set(CMAKE_CXX_STANDARD 14)
+set(CMAKE_CXX_STANDARD 17)
```

Comment:
Any reason for this? I would like to keep this the same as the one enforced by PyTorch.

Reply:
It's mainly for the convenience of tuple unpacking and `std::optional`. I could change them back by using `c10::optional` and manually unpack tuples.